piksi_tools/console/solution_view.py

```diff
@@ -127,72 +131,66 @@ def _paused_button_fired(self):
-    self.plot_data.set_data('alt_fixed', [])
+    self.plot_data.update_data({'cur_lat_spp': []})
+    self.plot_data.update_data({'cur_lng_spp': []})
+    self.plot_data.update_data({'cur_lat_dgnss': []})
+    self.plot_data.update_data({'cur_lng_dgnss': []})
+    self.plot_data.update_data({'cur_lat_float': []})
+    self.plot_data.update_data({'cur_lng_float': []})
+    self.plot_data.update_data({'cur_lat_fixed': []})
+    self.plot_data.update_data({'cur_lng_fixed': []})
+
+  def _zero_init(self):
+    self.lats = np.zeros(self.plot_history_max)
+    self.lngs = np.zeros(self.plot_history_max)
+    self.tows = np.zeros(self.plot_history_max)
+    self.modes = np.zeros(self.plot_history_max, dtype='int8')
 
   def _clear_button_fired(self):
-    self.tows = np.empty(self.plot_history_max)
-    self.lats = np.empty(self.plot_history_max)
-    self.lngs = np.empty(self.plot_history_max)
-    self.alts = np.empty(self.plot_history_max)
-    self.modes = np.empty(self.plot_history_max)
-    self._clear_history()
+    self._zero_init()
+    self.plot_data.update_data({'lat_spp': []})
+    self.plot_data.update_data({'lng_spp': []})
+    self.plot_data.update_data({'lat_dgnss': []})
+    self.plot_data.update_data({'lng_dgnss': []})
+    self.plot_data.update_data({'lat_float': []})
+    self.plot_data.update_data({'lng_float': []})
+    self.plot_data.update_data({'lat_fixed': []})
+    self.plot_data.update_data({'lng_fixed': []})
     self._reset_remove_current()
 
-
   def _pos_llh_callback(self, sbp_msg, **metadata):
     # Updating an ArrayPlotData isn't thread safe (see chaco issue #9), so
     # actually perform the update in the UI thread.
     if self.running:
       GUI.invoke_later(self.pos_llh_callback, sbp_msg)
   
   def age_corrections_callback(self, sbp_msg, **metadata):
-    age_msg = MsgAgeCorrections(sbp_msg)
-    if age_msg.age != 0xFFFF:
-      self.age_corrections = age_msg.age/10.0
+    if sbp_msg.age != 0xFFFF:
+      self.age_corrections = sbp_msg.age/10.0
     else:
       self.age_corrections = None
 
   def update_table(self):
     self.table = self.pos_table + self.vel_table + self.dops_table
 
   def auto_survey(self):
-    if self.last_soln.flags != 0: 
-      self.latitude_list.append(self.last_soln.lat)
-      self.longitude_list.append(self.last_soln.lon)
-      self.altitude_list.append(self.last_soln.height)
-    if len(self.latitude_list) > 1000:
-      self.latitude_list = self.latitude_list[-1000:]
-      self.longitude_list = self.longitude_list[-1000:]
-      self.altitude_list = self.altitude_list[-1000:]
-    if len(self.latitude_list) != 0:
-      self.latitude  = sum(self.latitude_list)/len(self.latitude_list)
-      self.altitude  = sum(self.altitude_list)/len(self.latitude_list)
-      self.longitude = sum(self.longitude_list)/len(self.latitude_list)
+    if 0 == self.last_soln.flags:
+      return
+
+    self.latitude_list = np.lib.pad(self.latitude_list,
```

Comment:
Ok now that I understood what the actual question was I looked into it more detail. So the original solution was that on each call to `update_on_heartbeat()` the lat, lon, alt references in settings_view were updated from solution_view. This of course works but is unnecessary. More matter of taste how you want to do it, I ended up passing solution_view pointer for settings_view. Now settings_view can use the variables from solution_view directly. To be pedant this should be protected by mutex since now there's a possibility that solution_view updates for example longitude_list right after settings_view has read latitude_list. This will cause one sample offset in calculations but since we are calculating mean for static position, effect is negligible.